Backendless/WebORB/Client/HttpEngineWithClient.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System;
+using System.Collections;
+using System.Threading;
+using System.Threading.Tasks;
+using Weborb.V3Types;
+using System.Net.Http;
+using Weborb.Message;
+using Weborb.Util;
+using Weborb.Reader;
+using System.Net;
+using System.Net.Http.Headers;
+using Weborb.Types;
+using Weborb.Protocols.Amf;
+using Weborb.Exceptions;
+
+namespace Weborb.Client
+{
+  public class HttpEngineWithClient : Engine
+  {
+    HttpClient httpClient = new HttpClient();
+    HttpWebRequest _request;
+    public HttpEngineWithClient( String url, IdInfo idInfo ) : base( url, idInfo )
+    {
+    }
+
+    public override void SendRequest<T>( V3Message v3Msg, IDictionary requestHeaders,
+                                         IDictionary httpHeaders, Responder<T> responder,
+                                         AsyncStreamSetInfo<T> asyncStreamSetInfo ) =>
+      ThreadPool.QueueUserWorkItem( state => SendHttpRequest( v3Msg, requestHeaders, httpHeaders, responder, asyncStreamSetInfo ) );
```

Comment:
@holailusoria is this needed? I thought httpclient already does the async processing, no?

Reply:
please investigate this, this is a very important line and I'd like you to have full understanding what it is for and whether it is needed